### ISSUE TITLE
Log type request.summary

### ISSUE THREAD

**Issue by leplatrem** [mozilla-services/go-mozlogrus#5]
I want to use `mozlogrus` to output a request summary.

A stated on the docs, https://wiki.mozilla.org/Firefox/Services/Logging, the `Type` should be set to `request.summary`.

The type seems to be hard-coded to `"app.log"` here.

I'd be happy to contribute something, but what would be the best way to specify it? Use `entry.Data["Type"]` if set?

Thanks

**Comment by leplatrem** [comment]
@oremj @mostlygeek any idea?

In Python the logger has a name:

```py
logger = logging.getLogger("request.summary")
logger.info("message")
```
Which then becomes the log type:
https://github.com/mozilla/mozilla-cloud-services-logger/blob/master/mozilla_cloud_services_logger/formatters.py#L46-L62

But I can't find the equivalent in logrus :/

**Comment by mostlygeek** [comment]
I'm bias. I don't use this library because I have issues with the side effects.

Instead, I wrote my [own http.Handler logger for for go-syncstorage](https://github.com/mozilla-services/go-syncstorage/blob/master/web/logHandler.go).  Maybe we can generalize that handler for other projects?  It has some static strings which are go-sync specific. Trivial to make as params. 

The side effects I have issues with: 

In the `Enable()` call it sets output to`stdout`.  That's not that big of a deal since you can avoid `Enable()` and do things manually.  Makes it impossible to test though. 

In the dependent library [go-mozlog's init()](https://github.com/mozilla-services/go-mozlog/blob/master/mozlog.go#L30-L39) call, it makes global changes to the `log` package's default logger :man_facepalming:.  Libraries making changes to globals is bad design.  It also makes writing tests really hard.  Since projects depend on these libraries and their behavior it's too late to change them.

**Comment by oremj** [comment]
I can't change go-mozlog's behavior, because it will break existing projects. Can you review #6? It removes the mozlog dependency and all init calls.


Addressing `Enable`, that is just a helper function for our applications, since they should log to stdout, but, as you mentioned, if other behavior is needed it should just be hooked up manually.

**Comment by leplatrem** [comment]
Thanks @oremj for the refactor!

> Since projects depend on these libraries and their behavior it's too late to change them.

There is no semver and changelogs in the Go world?

**Comment by mostlygeek** [comment]
there's no out of the box support. The idiomatic (but not official) way is simply have people use gopkg.in. Which a contributor runs. 

I pushed a v1.0.0 tag which can be accessible using: 

* `go get gopkg.in/mozilla-services/go-mozlogrus.v1`
* `go get gopkg.in/mozilla-services/go-mozlogrus.v1.0`
* `go get gopkg.in/mozilla-services/go-mozlogrus.v1.0.0`

Then we can make a v2 branch which fixes some of the side effects.  The problem will be the master branch which by default, `go get` fetches.  We can fix that later. For now we can just have a v2 branch.

**Comment by mostlygeek** [comment]
ok support for types has landed in #7.  You can enable logging with a custom type by using: 

```go
mozlogrus.EnableFormatter(&mozlogrus.MozLogFormatter{
        LoggerName: "ApplicationName",
        Type: "udp datagram",
})
```

**Comment by leplatrem** [comment]
Thanks!

I will probably remove the usage of the global `log` in my code I think...
I will instantiate several loggers (one with `type: "app.log"` for initialization messages etc, one for `type: "request.summary"`, and one for my particular use-case `type: "request.authz"`)
